File: C/SelectSort.c
void troca(int *a,int *b){
    int c;
    c=*a;
    *a=*b;
    *b=c;
}
void SelectSort(int *vetor,int inicio,int fim){
    int i,j,menor,posicao;
    for(i=inicio;i<(fim+1);i++){
        menor=*(vetor + i);
        posicao=i;
        for(j=i;j<(fim+1);j++){
            if(menor > *(vetor + j)){
                posicao=j;
                menor=*(vetor + j);
            }
            troca((vetor+i),(vetor+posicao));
        }
        
    }
}

File: Python/SelectSort.py
def SelectSort(vetor,inicio=0,fim=None):
    if fim == None:
        fim=len(vetor)-1
    fim+=1
    for i in range(inicio,fim):
        menor=vetor[i]
        posicao=i
        for j in range(i+1,fim):
            if menor > vetor[j]:
                menor=vetor[j]
                posicao=j
        vetor[i],vetor[posicao] = vetor[posicao],vetor[i]

        
        
